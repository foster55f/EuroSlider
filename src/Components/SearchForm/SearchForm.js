import React, { useState } from 'react';
import PropTypes from 'prop-types';
import App from '../App/App';
import { useSelector, useDispatch} from 'react-redux';


const SearchForm = (props) => {
    const games = useSelector(state => state.games);
    const [state, setState] = useState({foundGames:[], searchField:''});

    const handleSearch = (event) => {
        event.preventDefault()
        const foundGamesArray = games.filter(game => {
            return game.title.toLowerCase().includes(state.searchField.toLowerCase())|| game.title.toLowerCase().includes(state.searchField.toLowerCase())
        })
        props.search(foundGamesArray)
        setState({ searchField: '' })
    }

        return (
            <header>
            <div className='searchContainer'>   
                <input
                    className ='search-input'
                    type='text'
                    placeholder='Search For Highlights'
                    name='title'
                    onChange={event => setState({ searchField: event.target.value })}
                    value={state.searchField}
                    />
                    <div className='search-div'>
                        <button onClick={handleSearch} className="searchButton">Search</button>
                    </div>
                    
            </div>
        </header>
        )
    }


export default SearchForm

SearchForm.propTypes = {
    games: PropTypes.object,
}