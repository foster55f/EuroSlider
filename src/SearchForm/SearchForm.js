import React from 'react';
import { filterGames } from '../actions';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

class SearchForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            searchField: '',
            foundGames: []
        }
    }

    handleSearch = (event) => {
        event.preventDefault()
        const foundGamesArray = this.props.games.filter(game => {
            return game.title.toLowerCase().includes(this.state.searchField.toLowerCase())|| game.title.toLowerCase().includes(this.state.searchField.toLowerCase())
        })
        this.props.search(foundGamesArray)
        this.setState({ searchField: '' })
    }

   
    render() {
        return (
            <header>
            <div className='searchContainer'>   
                <input
                    className ='search-input'
                    type='text'
                    placeholder='Search For Highlights'
                    name='title'
                    onChange={event => this.setState({ searchField: event.target.value })}
                    value={this.state.searchField}
                />
                    <button onClick={this.handleSearch} className="searchButton">Search</button>
                    
            </div>
        </header>
        )
    }
}

export const mapStateToProps = (state) => ({
    games: state.games,
  })
  
  export const mapDispatchToProps = dispatch => ({
    filterGames: games => dispatch( filterGames(games) )
  })
  
  export default connect(mapStateToProps, mapDispatchToProps)(SearchForm)