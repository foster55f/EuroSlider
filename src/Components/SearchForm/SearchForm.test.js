import React from 'react';
import { shallow } from 'enzyme';
import { SearchForm, mapStateToProps} from './SearchForm';


describe('SearchForm', () => {
  let wrapper;

  it('should match the App snapshot', () => {
    wrapper = shallow(<SearchForm />);
    expect(wrapper).toMatchSnapshot();
  })


  describe('mapsStateToProps', () => {
    it('should return only games from the store', () => {
        const mockState = {
            games: []
        };
        const expected = {
            games:[]
        };
        const mappedProps = mapStateToProps(mockState);

        expect(mappedProps).toEqual(expected);
        });
    });
})