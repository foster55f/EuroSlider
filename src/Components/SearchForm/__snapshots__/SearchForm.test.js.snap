// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the App snapshot 1`] = `
<header>
  <div
    className="searchContainer"
  >
    <input
      className="search-input"
      name="title"
      onChange={[Function]}
      placeholder="Search For Highlights"
      type="text"
      value=""
    />
    <button
      className="searchButton"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</header>
`;
